pkg/policies/dataplane/response-metrics-registry.go

```diff
@@ -1,10 +1,146 @@
 package dataplane
 
+import (
+	"context"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"go.uber.org/fx"
+	"go.uber.org/multierr"
+
+	flowcontrolv1 "github.com/fluxninja/aperture/api/gen/proto/go/aperture/flowcontrol/v1"
+	"github.com/fluxninja/aperture/pkg/log"
+	"github.com/fluxninja/aperture/pkg/metrics"
+	"github.com/fluxninja/aperture/pkg/policies/dataplane/iface"
+)
+
 // TODO: Bring metrics to this file
 // 1. per component metrics from actuator, fluxmeter.
 
-// // ResponseMetricsGetAPI is an interface for getting response metrics.
-// type ResponseMetricsGetAPI interface {
-// 	GetFluxMeterHistogram(fluxMeterID string) (prometheus.Histogram, bool)
-// 	GetTokenLatencyHistogram(concurrencyControlID string) (prometheus.Histogram, bool)
-// }
+type responseMetricsRegistry struct {
+	pr                        *prometheus.Registry
+	workloadLatencySummaryVec *prometheus.SummaryVec
+	fluxmeterHistogramVec     *prometheus.HistogramVec
+}
+
+// ProvideResponseMetricsAPI returns API for getting metrics observers.
+func ProvideResponseMetricsAPI(registry *prometheus.Registry, lifecycle fx.Lifecycle) iface.ResponseMetricsAPI {
+	rm := &responseMetricsRegistry{
+		pr: registry,
+	}
+
+	// WORKLOAD LATENCY METRICS
+	rm.workloadLatencySummaryVec = prometheus.NewSummaryVec(prometheus.SummaryOpts{
+		Name: metrics.WorkloadLatencyMetricName,
+		Help: "Latency summary of workload",
+	}, []string{
+		metrics.PolicyNameLabel,
+		metrics.PolicyHashLabel,
+		metrics.ComponentIndexLabel,
+		metrics.DecisionTypeLabel,
+		metrics.WorkloadIndexLabel,
+	})
+
+	// FLUXMETER METRICS
+	rm.fluxmeterHistogramVec = prometheus.NewHistogramVec(prometheus.HistogramOpts{
```

Comment:
Each FluxMeter histogram has different buckets. Vector can represent only one set of buckets. Therefore, cannot use a single Vector to represent all FluxMeter metrics since buckets would change based on FluxMeter